src/models/meta_details.rs

```diff
@@ -355,6 +383,74 @@ fn streams_update<E: Env + 'static>(
     }
 }
 
+fn suggested_stream_update(
+    suggested_stream: &mut Option<Stream>,
+    selected: &Option<Selected>,
+    meta_items: &[ResourceLoadable<MetaItem>],
+    meta_streams: &[ResourceLoadable<Vec<Stream>>],
+    streams: &[ResourceLoadable<Vec<Stream>>],
+    stream_bucket: &StreamsBucket,
+) -> Effects {
+    let next_suggested_stream = match selected {
+        Some(Selected {
+            meta_path,
+            stream_path: Some(stream_path),
+        }) => {
+            meta_items
+                .iter()
+                .find_map(|meta_item| match &meta_item.content {
+                    Some(Loadable::Ready(meta_item)) => Some(&meta_item.videos),
+                    _ => None,
+                })
+                .and_then(|videos| {
+                    // Check saved stream only for last 30 videos starting from the current video
+                    videos
+                        .iter()
+                        .position(|video| video.id == stream_path.id)
+                        .and_then(|max_index| {
+                            videos[max_index.saturating_sub(30)..=max_index]
+                                .iter()
+                                .rev()
+                                .find_map(|video| {
+                                    stream_bucket.items.get(&StreamsItemKey {
+                                        meta_id: meta_path.id.to_string(),
+                                        video_id: video.id.to_owned(),
+                                    })
+                                })
+                        })
+                })
+                .and_then(|stream_item| {
+                    [meta_streams, streams]
+                        .concat()
+                        .iter()
+                        .find(|resource| resource.request.base == stream_item.stream_transport_url)
```

Comment:
Do we expect only 1 found match? If not then maybe instead of using `find` we could find with the match for the `resource.content` merging the two lines in order to find a `Loadable::Ready(streams)` using the `find` all

Reply:
ahhh that's addon URL not resource URL. Resolved then.